stm/stm.py

```diff
@@ -0,0 +1,418 @@
+                    out_shape=[out_x, out_y],
+                    fill=np.NAN,
+                    all_touched=True,
+                    default_value=1,
+                    transform=affine.Affine.from_gdal(
+                        min(self._obj[kwargs["x"]]),
+                        kwargs["dx"],
+                        0.0,
+                        max(self._obj[kwargs["y"]]),
+                        0.0,
+                        -1 * kwargs["dy"],
+                    ),
+                )
+                subset = [
+                    item for item in np.unique(raster) if not (math.isnan(item)) == True
+                ]
+                data_xr_subset = self._obj.sel(points=subset)
+            case "polygon":
+                _check_polygon_kwargs(**kwargs)
+                mask = self._obj.stm._in_polygon(
+                    kwargs["polygon"], xlabel="lon", ylabel="lat"
+                )
+                idx = self._obj.points.data[mask.data]
+                data_xr_subset = self._obj.sel(points=idx)
+            case other:
+                raise NotImplementedError(
+                    "Method: {} is not implemented.".format(method)
+                )
+        chunks = {
+            "points": min(
+                self._obj.chunksizes["points"][0], data_xr_subset.points.shape[0]
+            ),
+            "time": min(self._obj.chunksizes["time"][0], data_xr_subset.time.shape[0]),
+        }
+
+        data_xr_subset = data_xr_subset.chunk(chunks)
+
+        return data_xr_subset
+
+    def enrich_from_polygon(self, polygon, fields, xlabel="lon", ylabel="lat"):
+        """
+        Enrich the SpaceTimeMatrix from one or more attribute fields of a (multi-)polygon.
+
+        Each attribute in fields will be assigned as a data variable to the STM.
+        If a point of the STM falls into the given polygon, the value of the specified field will be added. For points outside the (multi-)polygon, the value will be None.
+
+        Parameters
+        ----------
+        polygon : geopandas.GeoDataFrame, str, or pathlib.Path
+            Polygon or multi-polygon with contextual information for enrichment
+        fields : str or list of str
+            Field name(s) in the (multi-)polygon for enrichment
+        xlabel : str, optional
+            Name of the x-coordinates of the STM, by default "lon"
+        ylabel : str, optional
+            Name of the y-coordinates of the STM, by default "lat"
+
+        Returns
+        -------
+        xarray.Dataset
+            Enriched STM.
+        """
+        _ = _validate_coords(self._obj, xlabel, ylabel)
+
+        # Check if fields is a Iterable or a str
+        if isinstance(fields, str):
+            fields = [fields]
+        elif not isinstance(fields, Iterable):
+            raise ValueError("fields need to be a Iterable or a string")
+
+        # Get polygon type and the first row
+        if isinstance(polygon, gpd.GeoDataFrame):
```

Comment:
adding support for WKT polygons (as strings) might be useful for users here

Reply:
converted to #18